aries_cloudagent/tails/indy_tails_server.py

```diff
@@ -30,9 +32,21 @@ async def upload_tails_file(
             backoff: exponential backoff in retry interval
             max_attempts: maximum number of attempts to make
         """
-
-        genesis_transactions = context.settings.get("ledger.genesis_transactions")
         tails_server_upload_url = context.settings.get("tails_server_upload_url")
+        if not (context.settings.get("ledger.genesis_transactions")):
+            if isinstance(context, Profile):
```

Comment:
Maybe we can update the calls to the `upload_tails_file` method to always pass the context? That way we don't have to do an instance check on the type

Reply:
Updated accordingly